src/components/postDetail/reservationPanel/Calendar/index.tsx

```diff
@@ -0,0 +1,152 @@
+import Image from 'next/image';
+
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+
+import classNames from 'classnames/bind';
+import dayjs from 'dayjs';
+
+import { SVGS } from '@/constants';
+
+import NoAvailableSchedule from '@/components/postDetail/reservationPanel/NoAvailableSchedule';
+import { availableSchedule } from '@/constants/mockData/reservationAvailableSchedule';
+import useToggleButton from '@/hooks/useToggleButton';
+
+import styles from './Calendar.module.scss';
+
+const cx = classNames.bind(styles);
+const { left, right } = SVGS.arrow;
+
+type CalenderProps = {
+  activityId: number;
+  setAvailableTimes: Dispatch<SetStateAction<AvailableTimesOptions[]>>;
+  setIsNoSchedule: Dispatch<SetStateAction<boolean>>;
+};
+
+type AvailableTimesOptions = {
+  title: string;
+  value: number;
+};
+
+const Calendar = ({ activityId, setAvailableTimes, setIsNoSchedule }: CalenderProps) => {
+  const today = dayjs();
+
+  const [selectedYear, setSelectedYear] = useState(today.year());
+  const [selectedMonth, setSelectedMonth] = useState(today.month() + 1);
+  const [selectedDate, setSelectedDate] = useState(availableSchedule[0]?.date || '');
+
+  const { isVisible: arrowLeftHover, handleToggleClick: arrowLeftClick } = useToggleButton();
+  const { isVisible: arrowRightHover, handleToggleClick: arrowRightClick } = useToggleButton();
+
+  const { url: arrowLeftUrl, alt: arrowLeftAlt } = arrowLeftHover ? left.active : left.default;
+  const { url: arrowRightUrl, alt: arrowRightAlt } = arrowRightHover ? right.active : right.default;
+  const isAllSchedulesReserved = availableSchedule.length === 0;
+  const monthName = dayjs()
+    .month(selectedMonth - 1)
+    .format('MMMM');
+  const maxDate = today.add(30, 'day');
+
+  const updateAvailableTimes = (value: string) => {
+    setSelectedDate(value);
+    const selectedDateSchedule = availableSchedule.find((a) => a.date === selectedDate);
+    let updatedTimes: AvailableTimesOptions[] = [];
+
+    if (selectedDateSchedule) {
+      updatedTimes = selectedDateSchedule.times.map((item) => ({
+        value: item.id,
+        title: `${item.startTime}-${item.endTime}`,
+      }));
+    } else {
+      updatedTimes = [
+        {
+          title: '예약 가능한 시간대가 없습니다.',
+          value: 0,
+        },
+      ];
+    }
+
+    setAvailableTimes(updatedTimes);
+  };
+
+  const handlePrevMonthClick = () => {
+    if (selectedMonth === today.month() + 1 && selectedYear === today.year()) return;
+
+    setSelectedMonth((prevMonth) => {
+      const newMonth = prevMonth === 1 ? 12 : prevMonth - 1;
+      const newYear = prevMonth === 1 ? selectedYear - 1 : selectedYear;
+      setSelectedYear(newYear);
+
+      return newMonth;
+    });
```

Comment:
setSelectedMonth 함수 안에서 setSelectedYear를 호출하는 것보단 아래처럼 변수를 빼놓고 각자 setState를 해줘도 되지 않을까요?

```suggestion
   const newMonth = selectedMonth === 1 ? 12 : selectedMonth- 1;
   const newYear = selectedMonth === 1 ? selectedYear - 1 : selectedYear;
   
   setSelectedMonth(newMonth);
   setSelectedYear(newYear);
```

Reply:
이전 상태값 기반해서 변경하도록 prev에 집착 했는데요 흠..... 바로 변경해도 상관 없겠지요? 
반영하겠습니다~

Reply:
넵넵 클릭할때만 동작하는 이벤트 핸들러라서 바로 변경해도 상관 없을거같습니다!
저도 캘린더 날짜 바꿀때 바로 변경했는데 잘 되더라구요👍🏼

Reply:
네! 감사합니다~